docs/messaging.rst

```diff
@@ -859,7 +859,9 @@ multiple cases:
 
 The client sends a shutdown request to the kernel, and once it receives the
 reply message (which is otherwise empty), it can assume that the kernel has
-completed shutdown safely.
+completed shutdown safely.  The request can be sent on either the `control` or
+`shell` channels.  The reply is made both to the requesting channel and to
+the `iopub` channel.
```

Comment:
But I'm pretty sure the R kernel, for instance, doesn't broadcast shutdown_reply on IOPub. And so far, it hasn't been a problem. So I want to work out whether we document what IPython currently does as the expectation, or leave it undocumented and possibly introduce a separate official shutdown broadcast message.